design/one-pager-mrds.md

```diff
@@ -0,0 +1,248 @@
+smaller providers][4]. This helped alleviate the issue, but some large
+Crossplane deployments still work around it - for example by building their own
+providers that have only the MRs they need.
+
+## Proposal
+
+I propose we introduce a new type - ManagedResourceDefinition (MRD).
+
+An MRD would be a lightweight abstraction on a CRD:
+
+```yaml
+apiVersion: apiextensions.crossplane.io/v1alpha1
+kind: ManagedResourceDefinition
+metadata:
+  name: nopresources.nop.crossplane.io
+spec:
+  group: nop.crossplane.io
+  names:
+    categories:
+    - nop
+    kind: NopResource
+    listKind: NopResourceList
+    plural: nopresources
+    singular: nopresource
+  scope: Cluster
+  versions:
+    name: v1alpha1
+    schema:
+      openAPIV3Schema:
+        # Omitted for brevity
+    served: true
+    storage: true
+  connectionDetails:
+  - name: password
+    description: Definitely real password for this resource that does nothing
+  desiredState: Active
+```
+
+An MRD would be schematically identical to a CRD, but have two additional
+`spec` fields:
+
+* `spec.connectionDetails` - An array of connection detail keys and descriptions
+* `spec.desiredState` - Toggles whether the underlying CRD is created or not
+
+I propose we update all providers to deliver MRDs - not CRDs - as their package
+payload.
+
+I propose all MRDs be inactive (`spec.desiredState: Inactive`) by default, and
+that we add another new type to automatically activate matching MRDs. This new
+type would be called a ManagedResourceActivationPolicy (MRAP).
+
+Here's an example MRAP:
+
+```yaml
+apiVersion: apiextensions.crossplane.io/v1alpha1
+kind: ManagedResourceActivationPolicy
+metadata:
+  name: aws
+spec:
+  activate:
+  - instances.rds.m.aws.crossplane.io
+  - *.ec2.m.aws.crossplane.io
+```
+
+An MRAP would specify an array of MRD names to activate. The array supports
+wildcard prefixes (like `*.aws.crossplane.io`) but not full regular expressions.
+
+A controller would watch MRAPs. Whenever any MRAP changes the controller would:
+
+1. List all MRAPs
+1. Compute the set of MRs that should be active
+1. Activate the relevant MRDs
+
+Crossplane would create a default MRAP at install time that activated `*` - i.e.
+all MRs. A CLI flag (and Helm chart value) would allow you to opt out of
+installing this default MRAP.
+
+Crossplane would support packaging MRAPs in Configuration packages. This
+addresses my concern that MR CRD filtering breaks Crossplane's package
+dependency model (see [the provider family design][4] for context). Using MRAPs
+a Configuration package can:
+
+1. Depend on a provider (e.g. provider-aws-ec2)
+1. Include an MRAP to specify specifically what MRs it needs
+
+This proposal requires updates to all providers. We use [controller-runtime][5]
+to build providers. controller-runtime assumes you'll add all controllers to a
+`Manager`, which will refuse to start if any of its controllers are missing
```

Comment:
It will also fail to sync the listers of a kind is missing from the cluster but it will try its hardest to list, but you will get a timeout eventually. Which is a different kind of problem related to the same thing.